src/controller/CHIPDeviceController.h

```diff
@@ -116,8 +140,12 @@ class DLL_EXPORT DeviceController : public SecureSessionMgrDelegate, public Pers
      * Init function to be used when there exists a device layer that takes care of initializing
      * System::Layer and InetLayer.
      */
-    CHIP_ERROR Init(NodeId localDeviceId, PersistentStorageDelegate * storageDelegate = nullptr,
-                    System::Layer * systemLayer = nullptr, Inet::InetLayer * inetLayer = nullptr);
+    CHIP_ERROR Init(NodeId localDeviceId, ControllerInitParams params);
+
+    [[deprecated("Use ControllerInitParams instead")]] CHIP_ERROR Init(NodeId localDeviceId,
```

Comment:
Please don't add this without re-enabling -Wdeprecated-declarations:
https://github.com/project-chip/connectedhomeip/blob/master/build/config/compiler/BUILD.gn#L147

(i.e., We should **not** use our own deprecated interfaces. To do so breaks any project that includes CHIP, enables -Wdeprecated-declarations, and prohibits non-fatal diagnostics using -Werror)

To be honest, I don't think we need to worry about using [[deprecated]] at all prior to 1.0.

Reply:
So maybe we can just left this function as is, with a comment to indicate future changes to Init should use `ControllerInitParams`?

Reply:
Sure, that sounds good. My concern is just breaking builds.